crates/polars-core/src/chunked_array/list/iterator.rs

```diff
@@ -188,6 +188,47 @@ impl ListChunked {
         unsafe { self.amortized_iter().for_each(f) }
     }
 
+    /// Zip with a `ChunkedArray` then apply a binary function `F` elementwise.
+    #[must_use]
+    pub fn zip_and_apply_amortized<'a, T, I, F>(&'a self, ca: &'a ChunkedArray<T>, mut f: F) -> Self
+    where
+        T: PolarsDataType,
+        &'a ChunkedArray<T>: IntoIterator<IntoIter = I>,
+        I: TrustedLen<Item = Option<T::Physical<'a>>>,
+        F: FnMut(Option<UnstableSeries<'a>>, Option<T::Physical<'a>>) -> Option<Series>,
+    {
+        if self.is_empty() {
+            return self.clone();
+        }
+        let mut fast_explode = self.null_count() == 0;
+        // SAFETY: unstable series never lives longer than the iterator.
+        let mut out: ListChunked = unsafe {
+            self.amortized_iter()
+                .zip(ca)
+                .map(|(opt_s, opt_v)| {
+                    let out = f(opt_s, opt_v);
+                    match out {
+                        Some(out) if out.is_empty() => {
+                            fast_explode = false;
+                            Some(out)
+                        },
+                        None => {
+                            fast_explode = false;
+                            out
+                        },
+                        _ => out,
+                    }
+                })
+                .collect_trusted()
```

Comment:
I think we should make this a bit more resilient to ambiguous dtypes. We can use `collect_ca_trusted_with_dtype` and accept the list inner type as argument of this function and then construct the  `DataType::List<inner>` with that.

In a later PR we can do the same with `apply_amortized`.

Reply:
Ok, let's leave it for now. Maybe we can take a look later. Want to have this in the release. :)